File: sdk/dart/lib/src/notifications.dart
import 'http_client.dart';

/// Notifications: manage notification preferences
class NotificationsResource {
  final HttpClient httpClient;
  NotificationsResource(this.httpClient);

  /// GET /api/notifications - List all notifications
  Future<Map<String, dynamic>> list([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/notifications', params);
  }

  /// GET /api/notifications/new - Return only new portal notifications
  Future<Map<String, dynamic>> listNew([Map<String, dynamic>? params]) async {
    return httpClient.get('/api/notifications/new', params);
  }

  /// PUT /api/notifications/{id}/ack - Mark notification as read
  Future<Map<String, dynamic>> acknowledge([Map<String, dynamic>? params]) async {
    return httpClient.put('/api/notifications/{id}/ack', params);
  }

}

File: sdk/rust/src/notifications.rs
//! Notifications: manage notification preferences

use std::sync::Arc;
use crate::http_client::HttpClient;

pub struct NotificationsResource {
    http: Arc<HttpClient>,
}

impl NotificationsResource {
    pub fn new(http: Arc<HttpClient>) -> Self { Self { http } }

    /// GET /api/notifications - List all notifications
    pub async fn list(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/notifications", None).await
    }

    /// GET /api/notifications/new - Return only new portal notifications
    pub async fn list_new(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.get("/api/notifications/new", None).await
    }

    /// PUT /api/notifications/{id}/ack - Mark notification as read
    pub async fn acknowledge(&self, params: Option<&serde_json::Value>) -> Result<serde_json::Value, crate::http_client::ApiError> {
        self.http.put("/api/notifications/{id}/ack", None).await
    }

}
